infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServerCache.java

```diff
@@ -27,7 +29,8 @@
     public LogUnitServerCache(LogUnitServerConfig config, StreamLog streamLog) {
         this.streamLog = streamLog;
         this.dataCache = Caffeine.newBuilder()
-                .<Long, ILogData>weigher((addr, logData) -> logData.getSizeEstimate())
+                .<Long, ILogData>weigher((addr, logData) -> logData.getSizeEstimate() +
```

Comment:
Can you replace this with a function call? For now, we suppose there is no performance impact if number of streams in a transaction is less than 10 tables? But if there are more tables, it may decrease the performance. Please put comments to the function and check how many streams in the map, if it is bigger than a threshold like 20, print a warn log.

Reply:
done